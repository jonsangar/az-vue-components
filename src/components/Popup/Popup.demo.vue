<template lang="pug">
  .popupDemo
    Demo(title="Uso Básico")
      az-row
        az-button(@click="zoom = true" size="small") zoom
        az-button(@click="fade = true" size="small") fade
        az-button(@click="flip = true" size="small") flip
        az-button(@click="door = true" size="small") door
        az-button(@click="slideDown = true" size="small") slideDown
        az-button(@click="slideUp = true" size="small") slideUp

    az-popup(v-model="zoom" @hide="zoom = false" bounce)
      | Un modal vue con animación zoom.
    az-popup(v-model="fade" transition="fade" @hide="fade = false" bounce)
      | Un modal vue con animación fade.
    az-popup(v-model="flip" transition="flip" @hide="flip = false" bounce)
      | Un modal vue con animación flip.
    az-popup(v-model="door" transition="door" @hide="door = false" bounce)
      | Un modal vue con animación door.
    az-popup(v-model="slideDown" transition="slideDown" @hide="slideDown = false" bounce)
      | Un modal vue con animación slideDown.
    az-popup(v-model="slideUp" transition="slideUp" @hide="slideUp = false" bounce)
      | Un modal vue con animación slideUp.


    Demo(title="Posición")
      az-row
        az-button(@click="top = true" size="small") top
        az-button(@click="right = true" size="small") right
        az-button(@click="bottom = true" size="small") bottom
        az-button(@click="left = true" size="small") left

    az-popup(v-model="top" @hide="top = false" position="top")
      | Un modal vue con animación.
    az-popup(v-model="right" @hide="right = false" position="right")
      | Un modal vue con animación.
    az-popup(v-model="bottom" @hide="bottom = false" position="bottom")
      | Un modal vue con animación.
    az-popup(v-model="left" @hide="left = false" position="left")
      | Un modal vue con animación.

    Demo(title="Abrir a pantalla completa")
      az-row
        az-button(@click="full = true" size="small") show
        az-button(@click="fullRight = true" size="small") full right

    az-popup(v-model="full" @hide="full = false" full)
      | Un modal vue con animación a pantalla completa.
    az-popup(v-model="fullRight" @hide="fullRight = false" position="right" full)
      | Un modal vue con animación a pantalla completa.

    Demo(title="Varios Modales")
      az-button(@click="some = true" size="small") Mostrar modal 1

    az-popup(v-model="some" @hide="some = false" bounce)
      .text-center
        .margin-bottom-half Modal 1
        az-button(@click="some1 = true" size="small") Mostrar modal 2

    az-popup(v-model="some1" @hide="some1 = false" bounce)
      .text-center
        .margin-bottom-half Modal 2
        az-button(@click="some2 = true" size="small") Mostrar modal 3

    az-popup(v-model="some2" @hide="some2 = false" bounce)
      | Modal 3


    Demo(title="Modales anidados")
      az-button(@click="nest = true" size="small") Mostrar modal 1

    az-popup(v-model="nest" @hide="nest = false" bounce)
      .text-center
        .margin-bottom-half Modal 1
        az-button(@click="nest1 = true" size="small") Mostrar modal 2

      az-popup(v-model="nest1" @hide="nest1 = false" bounce)
        .text-center
          .margin-bottom-half Modal 2
          az-button(@click="nest2 = true" size="small") Mostrar modal 3

        az-popup(v-model="nest2" @hide="nest2 = false" bounce)
          | Modal 3


    Demo(title="Scroll")
      az-button(@click="scroll = true" size="small") show

    az-popup(v-model="scroll" @hide="scroll = false" bounce)
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>
      | Un modal vue con animación y scroll. <br>

    Demo(title="Cerrar con Esc")
      az-button(@click="esc = true" size="small") show

    az-popup(v-model="esc" @hide="esc = false" closeOnEsc)
      | Un modal vue con animación y cierre con esc.

    Demo(title="Abrir en el body")
      az-button(@click="container = true" size="small") show

    az-popup(v-model="container" @hide="container = false" container="body")
      | Un modal vue con animación y renderizado en el body.


</template>

<script>
export default {
  data() {
    return {
      zoom: false,
      fade: false,
      flip: false,
      door: false,
      slideDown: false,
      slideUp: false,
      scroll: false,
      some: false,
      some1: false,
      some2: false,
      nest: false,
      nest1: false,
      nest2: false,
      esc: false,
      container: false,
      full: false,
      top: false,
      right: false,
      bottom: false,
      left: false,
      fullRight: false,
    };
  },
};
</script>

<style lang="scss">
body {
  height: 100%;
}

.az-popup__content {
  padding: simple();
}
</style>
