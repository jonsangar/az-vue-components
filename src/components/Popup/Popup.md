---
device: true
---

# Popup

### Instalación

```javascript
import { Popup } from 'didor';

Vue.use(Popup);
```

---

### Uso

#### Uso básico

``` html
<az-button @click="show = true">Mostrar modal</az-button>

<az-popup v-model="show" @hide="show = false">
  | Un modal vue con animación.
</az-popup>
```

``` js
export default {
  data() {
    return {
      show: false
    }
  }
}
```

#### Animaciones

``` html
<!-- Animación zoom - Default -->
<az-popup v-model="show" @hide="show = false" bounce>
  ...
</az-popup>

<!-- Animación fade -->
<az-popup v-model="show" transition="fade" @hide="show = false">
  ...
</az-popup>

<!-- Animación door -->
<az-popup v-model="show" transition="door" @hide="show = false" bounce>
  ...
</az-popup>
```

#### Posición

``` html
<az-popup v-model="show" position="top" @hide="show = false">
  ...
</az-popup>
```

#### Pantalla completa

``` html
<az-popup v-model="show" @hide="show = false" full>
  ...
</az-popup>
```

#### Superponer modales

Desde dentro de un modal podemos llamar a otro modal que se mostrará por encima.

>El orden en el que se superponen los modales depende del orden en el que se hayan definido en el HTML. Si este orden puede cambiar, te recomiendo que indiques un container (sección siguiente), ya que se irán añadiendo al final de ese container según los vayas instanciando.

``` html
<az-button @click="modal1 = true">Mostrar modal 1</az-button>

<!-- Modal 1 -->
<az-popup v-model="modal1" @hide="modal1 = false">
  | Modal 1
  <az-button @click="modal2 = true">Mostrar modal 2</az-button>
</az-popup>

<!-- Modal 2 -->
<az-popup v-model="modal2" @hide="modal2 = false">
  | Modal 2
  <az-button @click="modal3 = true">Mostrar modal 3</az-button>
</az-popup>

<!-- Modal 3 -->
<az-popup v-model="modal3" @hide="modal3 = false">
  | Modal 3
</az-popup>
```

#### Anidar modales

Un modal puede contener a su vez otros modales

``` html
<az-button @click="modal1 = true">Mostrar modal 1</az-button>

<!-- Modal 1 -->
<az-popup v-model="modal1" @hide="modal1 = false">
  | Modal 1
  <az-button @click="modal2 = true">Mostrar modal 2</az-button>

  <!-- Modal 2 -->
  <az-popup v-model="modal2" @hide="modal2 = false">
    | Modal 2
    <az-button @click="modal3 = true">Mostrar modal 3</az-button>

    <!-- Modal 3 -->
    <az-popup v-model="modal3" @hide="modal3 = false">
      | Modal 3
    </az-popup>
  </az-popup>
</az-popup>
```

#### Abrir en el body

``` html
<az-popup v-model="show" container="body" @hide="show = false">
  ...
</az-popup>
```

---

### API Popup

| Attribute   | Description                                | Type      | Default   |
| ----------- | ------------------------------------------ | --------- | --------- |
| v-model     | Identifica si se muestra el popup          | `Boolean` | `false`   |
| mask        | Indica si se muestra una máscara de fondo  | `Boolean` | `false`   |
| closeButton | Indica si se muestra el botón de cerrar    | `Boolean` | `true`    |
| closeOnEsc  | Indica si el popup se cierra al pulsar la tecla `esc` | `Boolean` | `false` |
| closeOnClickMask | Indica si se cierra el popup al hacer click en la máscara | `Boolean` | `fatruese` |
| z-index     | Permite el valor del z-index               | `Number`  | `100`     |
| lockScroll  | Bloquea el scroll en el body               | `Boolean` | `true`    |
| container   | Indica el bloque html donde se incluirá el modal | `String`  | -- |
| position    | Indica en que lado se renderizará el modal, por defecto en el centro de la pantalla<br>`top` `right` `bottom` `left` | `String` | -- |
| full        | Indica si el modal debe ocupar toda la pantalla | `Boolean` | `false` |
| duration    | Indica la duración de la animación en milisegundos | `Number` | `300` |
| transition  | Indica el tipo de animación<br>`zoom` `fade` `flip` `door` `slideDown` `slideUp` | `String` | `zoom` |
| bounce      | Indica si la animación debe tener rebote   | `Boolean` | `false` |

### Eventos Popup

| Evento | Descripción                                             | Argumento |
| ------ | ------------------------------------------------------- | --------- |
| hide   | Se lanza cuando se cierra el Popup y cuando se hace click sobre la máscara |          |
| clickMask | Se lanza cuando se hace click en la máscara          |          |

### Slots

| Nombre | Descripción                                             |
| ------ | ------------------------------------------------------- |
| default | Contenido del modal |
