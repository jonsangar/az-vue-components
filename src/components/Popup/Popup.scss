.az-overflow-hidden {
  overflow: hidden !important;

  // > *:not(.az-popup) {
  //   filter: blur(3px) grayscale(50%);
  // }
}

.az-popup,
.az-popup__mask {
  top: 0;
  left: 0;
  z-index: 100;
  width: 100%;
  height: 100%;
}

.az-popup {
  position: fixed;
  display: flex;
  align-items: center;
  justify-content: center;

  &__mask {
    position: absolute;
    background: rgba(0, 0, 0, 0.6);
  }

  &__dialog {
    @include shadow(3);

    position: relative;
    z-index: 101;
    overflow: hidden;
    background: color(background);
    border-radius: $border-radius;
  }

  &__content {
    max-width: 90vw;
    max-height: 90vh;
    overflow: auto;
  }

  &--top,
  &--bottom,
  &--left,
  &--right,
  &--full {
    .az-popup__content {
      max-width: initial;
      max-height: initial;
    }
  }

  &--top {
    align-items: flex-start;

    .az-popup__dialog {
      border-radius: 0 0 $border-radius $border-radius;
    }

    .az-popup__content {
      width: 100vw;
    }
  }

  &--bottom {
    align-items: flex-end;

    .az-popup__dialog {
      border-radius: $border-radius $border-radius 0 0;
    }

    .az-popup__content {
      width: 100vw;
    }
  }

  &--left {
    justify-content: flex-start;

    .az-popup__dialog {
      border-radius: 0 $border-radius $border-radius 0;
    }

    .az-popup__content {
      height: 100vh;
    }
  }

  &--right {
    justify-content: flex-end;

    .az-popup__dialog {
      border-radius: $border-radius 0 0 $border-radius;
    }

    .az-popup__content {
      height: 100vh;
    }
  }

  &--full {
    .az-popup__dialog {
      border-radius: 0;
    }

    .az-popup__content {
      width: 100vw;
      height: 100vh;
    }
  }
}

.az-popup__close {
  position: absolute;
  top: 0;
  right: 0;
  z-index: 100;
  width: double();
  height: double();
  cursor: pointer;
}

.az-popup__close::before,
.az-popup__close::after {
  position: absolute;
  top: 50%;
  left: 50%;
  width: half();
  height: 2px;
  content: '';
  background: color(gray1);
  border-radius: 100%;
  transition: background 0.2s;
}

.az-popup__close::before {
  transform: translate(-50%, -50%) rotate(45deg);
}

.az-popup__close::after {
  transform: translate(-50%, -50%) rotate(-45deg);
}

.az-popup__close:hover::before,
.az-popup__close:hover::after {
  background: color(brand);
}

//
// Fade animation
//
.az-popup-none-active {
  animation: az-popup-none-enter both ease-in;
}

.az-popup-none-leave-active {
  animation: az-popup-none-leave both ease-out;
}

@keyframes az-popup-none-enter {
  from {
    opacity: 1;
  }
}

@keyframes az-popup-none-leave {
  to {
    opacity: 1;
  }
}

//
// Fade animation
//
.az-popup-fade-enter-active {
  animation: az-popup-fade-enter both ease-in;
}

.az-popup-fade-leave-active {
  animation: az-popup-fade-leave both ease-out;
}

@keyframes az-popup-fade-enter {
  from {
    opacity: 0;
  }
}

@keyframes az-popup-fade-leave {
  to {
    opacity: 0;
  }
}

//
// Zoom animation
//
.az-popup-zoom-enter-active {
  animation: az-popup-zoom-enter both;
}

.az-popup-zoom-leave-active {
  animation: az-popup-zoom-leave both;
}

@keyframes az-popup-zoom-enter {
  from {
    opacity: 0;
    transform: scale3d(0.3, 0.3, 0.3);
  }
}

@keyframes az-popup-zoom-leave {
  to {
    opacity: 0;
    transform: scale3d(0.3, 0.3, 0.3);
  }
}

//
// Flip animation
//
.az-popup-flip-enter-active {
  animation: az-popup-flip-enter both ease-in;
  backface-visibility: visible !important;
}

.az-popup-flip-leave-active {
  animation: az-popup-flip-leave both;
  backface-visibility: visible !important;
}

@keyframes az-popup-flip-enter {
  from {
    opacity: 0;
    transform: perspective(400px) rotate3d(1, 0, 0, 60deg);
  }
  70% {
    transform: perspective(400px) rotate3d(1, 0, 0, -15deg);
  }
  to {
    transform: perspective(400px);
  }
}

@keyframes az-popup-flip-leave {
  from {
    transform: perspective(400px);
  }
  30% {
    transform: perspective(400px) rotate3d(1, 0, 0, -15deg);
  }
  to {
    opacity: 0;
    transform: perspective(400px) rotate3d(1, 0, 0, 45deg);
  }
}

//
// slideDown animation
//
.az-popup-slideDown-enter-active {
  animation: az-popup-slideDown-enter both;
}

.az-popup-slideDown-leave-active {
  animation: az-popup-slideDown-leave both;
}

@keyframes az-popup-slideDown-enter {
  from {
    opacity: 0;
    transform: translate3d(0, -100px, 0);
  }
}

@keyframes az-popup-slideDown-leave {
  to {
    opacity: 0;
    transform: translate3d(0, -100px, 0);
  }
}

//
// slideUp animation
//
.az-popup-slideUp-enter-active {
  animation: az-popup-slideUp-enter both;
}

.az-popup-slideUp-leave-active {
  animation: az-popup-slideUp-leave both;
}

@keyframes az-popup-slideUp-enter {
  from {
    opacity: 0;
    transform: translate3d(0, 100px, 0);
  }
}

@keyframes az-popup-slideUp-leave {
  to {
    opacity: 0;
    transform: translate3d(0, 100px, 0);
  }
}

//
// door animation
//
.az-popup-door-enter-active {
  animation: az-popup-door-enter both;
}

.az-popup-door-leave-active {
  animation: az-popup-door-leave both;
}

@keyframes az-popup-door-enter {
  from {
    opacity: 0;
    transform: scale3d(0, 1, 1);
  }
}

@keyframes az-popup-door-leave {
  60% {
    transform: scale3d(0.01, 1, 1);
  }
  to {
    opacity: 0;
    transform: scale3d(0, 1, 0.1);
  }
}

//
// top animation
//
.az-popup-top-enter-active,
.az-popup-top-leave-active {
  transition: transform;
}

.az-popup-top-enter,
.az-popup-top-leave-to {
  transform: translate3d(0, -100%, 0);
}

//
// bottom animation
//
.az-popup-bottom-enter-active,
.az-popup-bottom-leave-active {
  transition: transform;
}

.az-popup-bottom-enter,
.az-popup-bottom-leave-to {
  transform: translate3d(0, 100%, 0);
}

//
// left animation
//
.az-popup-left-enter-active,
.az-popup-left-leave-active {
  transition: transform;
}

.az-popup-left-enter,
.az-popup-left-leave-to {
  transform: translate3d(-100%, 0, 0);
}

//
// right animation
//
.az-popup-right-enter-active,
.az-popup-right-leave-active {
  transition: transform;
}

.az-popup-right-enter,
.az-popup-right-leave-to {
  transform: translate3d(100%, 0, 0);
}
