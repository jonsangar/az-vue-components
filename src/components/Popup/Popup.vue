<template lang="pug">
  transition(name="az-popup-none")
    .az-popup(
      v-show="value"
      tabindex="-1"
      :style="popupStyle"
      :class="popupClass"
      @keyup.esc="onEsc")

      //- Modal Mask
      transition(name="az-popup-fade")
        .az-popup__mask(v-show="mask && value" :style="popupStyle" @click="onClickMask")

      //- Modal Body
      transition(:name="`az-popup-${animation}`")
        .az-popup__dialog(v-show="value" :style="dialogStyle")

          //- Botón cerrar
          span.az-popup__close(v-if="closeButton" @click="$emit('hide')")

          //- Contenido
          .az-popup__content(@click="onClickContent")
            slot
</template>

<script>

export default {
  props: {
    value: {
      type: [Boolean, String, Number],
      default: true,
    },
    duration: {
      type: Number,
      default() {
        return this.$DIDOR ? this.$DIDOR.popup.duration : 300;
      },
    },
    transition: {
      type: String,
      default() {
        return this.$DIDOR ? this.$DIDOR.popup.transition : 'zoom';
      },
    },
    mask: {
      type: Boolean,
      default() {
        return this.$DIDOR ? this.$DIDOR.popup.mask : true;
      },
    },
    closeButton: {
      type: Boolean,
      default() {
        return this.$DIDOR ? this.$DIDOR.popup.closeButton : true;
      },
    },
    closeOnEsc: {
      type: Boolean,
      default() {
        return this.$DIDOR ? this.$DIDOR.popup.closeOnEsc : false;
      },
    },
    closeOnClickMask: {
      type: Boolean,
      default() {
        return this.$DIDOR ? this.$DIDOR.popup.closeOnClickMask : false;
      },
    },
    zIndex: {
      type: Number,
      default() {
        return this.$DIDOR ? this.$DIDOR.popup.zIndex : 100;
      },
    },
    lockScroll: {
      type: Boolean,
      default() {
        return this.$DIDOR ? this.$DIDOR.popup.lockScroll : true;
      },
    },
    container: {
      type: String,
      default() {
        return this.$DIDOR ? this.$DIDOR.popup.container : null;
      },
    },
    position: {
      type: String,
      default: '',
      validator: value => ['top', 'right', 'bottom', 'left', ''].indexOf(value) !== -1,
    },
    full: {
      type: Boolean,
      default: false,
    },
    bounce: {
      type: Boolean,
      default() {
        return this.$DIDOR ? this.$DIDOR.popup.bounce : false;
      },
    },
  },

  data() {
    return {
      opened: false,
    };
  },

  watch: {
    value(val) {
      const type = val ? 'open' : 'close';
      this[type](); // Llama al método => this.open() o this.close()
      this.$emit(type); // Emite un evento 'open' o 'close'
    },
  },

  computed: {
    popupClass() {
      return [this.position ? `az-popup--${this.position}` : '', this.full ? 'az-popup--full' : ''];
    },

    popupStyle() {
      return {
        animationDuration: `${this.duration}ms`,
        'z-index': this.zIndex,
      };
    },

    dialogStyle() {
      return {
        transitionDuration: `${this.duration}ms`,
        animationDuration: `${this.duration}ms`,
        transitionTimingFunction: this.bounce
          ? 'cubic-bezier(0.4, 0, 0, 1.5)'
          : 'cubic-bezier(0.55, 0, 0.1, 1)',
        animationTimingFunction: this.bounce
          ? 'cubic-bezier(0.4, 0, 0, 1.5)'
          : 'cubic-bezier(0.55, 0, 0.1, 1)',
      };
    },

    animation() {
      return this.position ? this.position : this.transition;
    },
  },

  methods: {
    open() {
      if (this.opened) return;

      this.opened = true;

      if (this.lockScroll) {
        document.body.classList.add('az-overflow-hidden');
      }
    },

    close() {
      if (!this.opened) return;

      if (this.lockScroll) {
        document.body.classList.remove('az-overflow-hidden');
      }

      this.opened = false;
      this.$emit('input', false);
    },

    onEsc() {
      if (this.value && this.closeOnEsc) {
        this.$emit('hide');
      }
    },

    onClickMask() {
      this.$emit('clickMask');
      if (this.closeOnClickMask) {
        this.$emit('hide');
      }
    },

    onClickContent(e) {
      // console.log('onClickContent: ', e)
      if (e?.target?.href){
        e.preventDefault()
        this.$emit('openlink', e.targe.href)
      }
    },
  },

  mounted() {
    // console.log('$DIDOR');
    // console.log(this.$DIDOR);
    if (this.container) {
      const element = document.querySelector(this.container);
      if (element) {
        element.appendChild(this.$el);
      } else {
        console.error(`az-popup: No se ha encontrado el elemento: ${this.container}`);
      }
    }

    if (this.value) {
      this.open();
    }
  },

  beforeDestroy() {
    this.close();
  },
};
</script>

<style src="./Popup.scss" lang="scss"></style>
