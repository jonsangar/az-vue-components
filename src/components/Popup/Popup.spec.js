import { shallowMount, createLocalVue } from '@vue/test-utils';
// Importamos librerías de estilos e iconos
import SvgIcon from 'vue-svgicon'
import Popup from './Popup.vue';

const localVue = createLocalVue()
localVue.use(SvgIcon, {
  tagName: 'az-icon',
  classPrefix: 'az',
})

describe('Popup.vue', () => {
  const build = () => {
    const wrapper = shallowMount(Popup, {
      localVue,
      propsData: {
        value: true
      }
    })

    return {
      wrapper,
    }
  }

  it('Comprueba renderizado del componente Popup con snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Comprueba el validator para la posición "position"', async () => {
    // NOTA: Los validadores es mejor testearlos de forma aislada (sin montar);
    const validator = Popup.props.position.validator;

    // Testeamos el validador de la prop como una función normal
    expect(validator('left')).toBe(true);
    expect(validator('right')).toBe(true);
    expect(validator('bottom')).toBe(true);
    expect(validator('top')).toBe(true);
    expect(validator('')).toBe(true);
    expect(validator('fakeposition')).toBe(false);
  });
})
