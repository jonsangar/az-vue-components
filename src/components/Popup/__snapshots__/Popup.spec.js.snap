// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup.vue Comprueba renderizado del componente Popup con snapshot 1`] = `
<transition-stub name="az-popup-none">
  <div tabindex="-1" class="az-popup" style="animation-duration: 300ms; z-index: 100;">
    <transition-stub name="az-popup-fade">
      <div class="az-popup__mask" style="animation-duration: 300ms; z-index: 100;"></div>
    </transition-stub>
    <transition-stub name="az-popup-zoom">
      <div class="az-popup__dialog" style="transition-duration: 300ms; animation-duration: 300ms; transition-timing-function: cubic-bezier(0.55, 0, 0.1, 1); animation-timing-function: cubic-bezier(0.55, 0, 0.1, 1);"><span class="az-popup__close"></span>
        <div class="az-popup__content"></div>
      </div>
    </transition-stub>
  </div>
</transition-stub>
`;
